### ISSUE TITLE
 download-from-binary-cache.pl: still waiting for 

### ISSUE THREAD

**Issue by gitfoxi** [NixOS/nix#805]
download-from-binary-cache.pl: still waiting for ‘https://cache.nixos.org/2r51pwshg93qqi58qz9r43iqvvnnnp89.narinfo’ after 5 seconds...

Is this supposed to make it faster? Because it makes it slower.

**Comment by edolstra** [comment]
Closing, not clear what issue is being described.

**Comment by gitfoxi** [comment]
Seems like it always times out.

**Comment by lschuermann** [comment]
I personally experience the same error quite often when interacting with the NixOS caching server.
Sadly, #14874 escalated a little and was closed (for reasons I fully understand) although it is a real issue for some people including me.

I noticed, that the error above pretty much only occurs on IPv6 enabled connections and disabling v6 often seems help. Is there any way to disable IPv6 connections with the `nix-env` (etc.) commands?
I don't want to keep keep IPv6 disabled on my DualStack connection, as I've got important v6-only services running that I then can't access anymore.
